### PR TITLE
Fix computation of OWA

### LINKED ISSUE #1556: M4 competition metric "OWA" is computed incorrectly
## Description

It seems like GluonTS computes the M4 competition metric "OWA" incorrectly. GluonTS computes the OWA per time-series and then averages it. However, the [competition guide](https://mk0mcompetitiont8ake.kinstacdn.com/wp-content/uploads/2018/09/M4-Competitors-Guide.pdf) states that the OWA is only computed on the final sMAPE and MASE values:

> Note that MASE and sMAPE are first estimated per series by averaging the error computed per forecasting horizon and then averaged again across the 3003 time series to compute their value for the whole dataset. On the other hand, OWA is computed only once at the end for the whole sample.

### PR DESCRIPTION
**Pull request by borchero**
Fixes #1556 

*Description of changes:*

- For each time series, only compute the sMAPE and MASE of Naive2
- Compute the OWA only for the final stats

**TBD:** new test

### CODE REVIEW

File: src/gluonts/evaluation/_base.py [hunk 1]
```diff
@@ -31,6 +31,10 @@
 import numpy as np
 import pandas as pd
 
+from gluonts.gluonts_tqdm import tqdm
+from gluonts.model.forecast import Forecast, Quantile
+from gluonts.model.naive_2 import naive_2
```
Comment (jaheba) [inline src/gluonts/evaluation/_base.py:36]: This requires `statsmodules` which is not installed by default

Comment (borchero) [reply src/gluonts/evaluation/_base.py]: Changed where it's imported

**Review by vafl — APPROVED** [review]
Thanks @borchero !

It would be nice to add a small test. Maybe based on the example in the m4 [guide lines](https://mk0mcompetitiont8ake.kinstacdn.com/wp-content/uploads/2018/09/M4-Competitors-Guide.pdf), but not urgent. I can also put it in the backlog.

File: src/gluonts/evaluation/_base.py [hunk 2]
```diff
@@ -461,6 +466,25 @@ def get_aggregate_metrics(
                 for q in self.quantiles
             ]
         )
+
+        # Compute OWA if required
+        if self.calculate_owa:
+            if totals["sMAPE_naive2"] == 0 or totals["MASE_naive2"] == 0:
+                warnings.warn(
+                    "OWA cannot be computed as Naive2 yields an sMAPE or MASE of 0."
+                )
```
Comment (lostella) [inline src/gluonts/evaluation/_base.py:475]: Elsewhere we use the `logging` module for warnings, would it make sense to also use it here?

**Review by lostella — APPROVED** [review]
Looks good! Thanks for spotting this!

**Comment by kashif** [comment]
awesome find @borchero 😎

**Pull request merged**